pkg/aws/awscontrolplane/mutate_awscontrolplane.go

```diff
@@ -66,6 +80,42 @@ func (m *Mutator) Mutate(request *admissionv1.AdmissionRequest) ([]mutator.Patch
 	if err != nil {
 		return nil, microerror.Maskf(parsingFailedError, "unable to parse release version from AWSControlPlane")
 	}
+
+	patch, err = m.MutateAWSControlPlane(request, *awsControlPlaneCR, releaseVersion)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	result = append(result, patch...)
+
+	return result, nil
+}
+
+// MutateUpdate is the function executed for every update webhook request.
+func (m *Mutator) MutateUpdate(request *admissionv1.AdmissionRequest) ([]mutator.PatchOperation, error) {
+	var result []mutator.PatchOperation
+	var patch []mutator.PatchOperation
+	var err error
+
+	awsControlPlaneCR := &infrastructurev1alpha2.AWSControlPlane{}
+	if _, _, err := mutator.Deserializer.Decode(request.Object.Raw, nil, awsControlPlaneCR); err != nil {
+		return nil, microerror.Maskf(parsingFailedError, "unable to parse awscontrol plane: %v", err)
+	}
+	releaseVersion, err := releaseVersion(awsControlPlaneCR)
+	if err != nil {
+		return nil, microerror.Maskf(parsingFailedError, "unable to parse release version from AWSControlPlane")
+	}
+
+	patch, err = m.MutateAWSControlPlane(request, *awsControlPlaneCR, releaseVersion)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	result = append(result, patch...)
+
+	return result, nil
+}
+
+func (m *Mutator) MutateAWSControlPlane(request *admissionv1.AdmissionRequest, awsControlPlaneCR infrastructurev1alpha2.AWSControlPlane, releaseVersion *semver.Version) ([]mutator.PatchOperation, error) {
```

Comment:
this is our temporary "all the logic goes here" function that will be split up next.